### PR TITLE
Added Demonfire and Cackling Flames.

### PR DESCRIPTION
**Pull request by JotaPeRL**

### CODE REVIEW

File: Mage.Sets/src/mage/sets/dissension/CacklingFlames.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+public class CacklingFlames extends CardImpl {
+
+    public CacklingFlames(UUID ownerId) {
+        super(ownerId, 59, "Cackling Flames", Rarity.COMMON, new CardType[]{CardType.INSTANT}, "{3}{R}");
+        this.expansionSetCode = "DIS";
+        
+        this.color.setRed(true);
+
+        // Cackling Flames deals 3 damage to target creature or player.
+        this.getSpellAbility().addEffect(new ConditionalOneShotEffect(
```
Comment (FenrisulfrX) [inline Mage.Sets/src/mage/sets/dissension/CacklingFlames.java]: You can simplify this card by using this constructor instead.
ConditionalOneShotEffect(OneShotEffect effect, OneShotEffect otherwiseEffect, Condition condition, String text)

**Comment by JotaPeRL** [comment]
damageTargetEffect() is how Blaze was implemented. I used it as template. It was also used in Crater's Claws. I thought they were similar enough.

**Comment by LevelX2** [comment]
> damageTargetEffect() is how Blaze was implemented. I used it as template. It was also used in Crater's Claws. I thought they were similar enough.

But you implement the following rule:
 // Hellbent - If you have no cards in hand, Demonfire can't be countered by spells or abilities and the damage can't be prevented.
There is no damage included. You added the damage effect twice, didn't you?

**Comment by JotaPeRL** [comment]
I added twice because in the second case (hellbent) it can't be prevented. It's a similar approach you did with Arrow Storm, that I also used as a template.
I think I'm not following your train of thought here. I'd like to fix the implementation, but I'm not sure what's the issue.

**Comment by LevelX2** [comment]
You're right, the implementation is correct. Sorry about the confusion.

**Pull request merged**